terraform/monitoring/02_dashboards.tf

```diff
@@ -114,4 +114,51 @@ resource "google_monitoring_dashboard" "userexp_dashboard" {
   }
 }
 EOF
+}
+
+# Here we create generic dashboards for three of the microservices. Since all three microservices
+# will share the same charts across their dashboards, we can leverage Terraform template files
+# in order to reproduce identical dashboards for each microservice.
+
+# Here we define specifics for each of the services that will receive a generic dashboard.
+# We need the service name for dashboard and chart titles, and we need the service ID to use as a metadata filter.
+variable "services" {
+  type = list(object({
+    service_name = string,
+    service_id = string
+  }))
+  default = [
+    {
+      service_name = "Payment Service"
+      service_id = "paymentservice"
+    },
+    {
+      service_name = "Email Service"
+      service_id = "emailservice"
+    },
+    {
+      service_name = "Shipping Service"
+      service_id = "shippingservice"
+    }
+  ]
+}
+
+# Here we iterate over the services that we defined and create a dashboard template file for each one using
```

Comment:
nit: remove "Here we" everywhere - focus on the action: "iterate over the services..."

Reply:
Fixed :)